src/smtp.rs

```diff
@@ -107,6 +107,127 @@ impl Smtp {
         .await
     }
 
+    async fn connect_secure_socks5(
+        &self,
+        context: &Context,
+        hostname: &str,
+        port: u16,
+        strict_tls: bool,
+        socks5_config: Socks5Config,
```

Comment:
Maybe introduce some `enum Transport` with variants `Tcp`, `Socks5(Socks5Config)`? This way we can collapse these 6 functions into 3 and have less, but more generic code

Reply:
Into a separate module maybe?
Ok, let's postpone the refactoring